services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -4801,6 +4801,81 @@ public Map<String, StoreDataAudit> getClusterStaleStores(String clusterName) {
     return retMap;
   }
 
+  /**
+   * This function iterates over all of Helix Parent Admin's child controllers,
+   * in order to obtain the stores and respective store information of each child controller,
+   * and then filter out the stores that are ready for compaction.
+   * @param clusterName, the name of the cluster to search for stores that are ready for compaction
+   * @return a list of StoreInfo objects of stores that are ready for compaction
+   */
+  @Override
+  public ArrayList<StoreInfo> getStoresForCompaction(String clusterName) {
+    ArrayList<StoreInfo> compactionReadyStores = new ArrayList<>();
+    try {
+      Map<String, ControllerClient> childControllers = getVeniceHelixAdmin().getControllerClientMap(clusterName);
+      ArrayList<StoreInfo> storeInfoList = new ArrayList<>();
+
+      // iterate through child controllers
+      for (Map.Entry<String, ControllerClient> controller: childControllers.entrySet()) {
+
+        // add all store info to storeInfoList
+        MultiStoreInfoResponse response = controller.getValue().getClusterStores(clusterName);
+        storeInfoList.addAll(response.getStoreInfoList());
+      }
+
+      // filter out
+      filterStoresForCompaction(storeInfoList, compactionReadyStores);
+    } catch (Exception e) {
+      throw new VeniceException("Something went wrong trying to fetch stores for compaction.", e);
+    }
+    return compactionReadyStores;
+  }
+
+  // package exclusive for testing
+  void filterStoresForCompaction(ArrayList<StoreInfo> storeInfoList, ArrayList<StoreInfo> compactionReadyStores) {
+    for (StoreInfo storeInfo: storeInfoList) {
+      if (isCompactionReady(storeInfo)) {
+        compactionReadyStores.add(storeInfo);
+      }
+    }
+  }
+
+  // This function abstracts the criteria for a store to be ready for compaction
+  private boolean isCompactionReady(StoreInfo storeInfo) {
+    return isHybridStore(storeInfo) && isLastCompactionTimeOlderThanThresholdHours(24, storeInfo);
+    // TODO: code style: store compactionThresholdHours as final var/enum
+  }
+
+  // START isCompactionReady() helper methods: each method below encapsulates a log compaction readiness criterion
+  private boolean isHybridStore(StoreInfo storeInfo) {
+    return storeInfo.getHybridStoreConfig() != null;
+  }
```

Comment:
While I was the one to suggest modularization. Feel free to make the call on breaking methods; E.g., `storeInfo.getHybridStoreConfig() != null` is probably simple enough to wrap inside the `isCompactionReady` method. We can always break the methods if the logic identifying hybrid stores evolves.

It is more of a subjective comment. Feel free to leave this as is or merge it together.

Reply:
Thank you for your suggestion. I had the same thought process and decided to modularise for readability.